sympy/core/tests/test_arit.py

```diff
@@ -443,12 +443,12 @@ def test_Add_Mul_is_finite():
     x = Symbol('x', extended_real=True, finite=False)
 
     assert sin(x).is_finite is True
-    assert (x*sin(x)).is_finite is None
+    assert (x*sin(x)).is_finite is False
```

Comment:
Here it is not obvious what should be returned.

`(x/S.Pi).is_integer` returns `False`, but can `"sin(oo)"` still possibly be zero? And is `oo*0 (= NaN)` in that case finite?

Reply:
It's correct that `(x/S.Pi).is_integer` should give `False` because an integer is always finite. Perhaps something somewhere should be checking for `finite` or perhaps more precisely `real`.

I think that `(x*sin(x)).is_finite` should be `None`.

Reply:
How can it possibly be finite if x is infinite?

Reply:
You can have something well defined like:
```python
In [1]: limit(x*sin(1/x**2), x, oo)
Out[1]: 0
```
Here the limit is:
```python
In [1]: limit(x*sin(x), x, oo)
Out[1]: ∞⋅sign(<-1, 1>)
```
which is a round about way of saying that in the limit this ends up stretching over all possible real numbers including zero.

It's important that the assumptions system gives `None` in any case that is ambiguous. More complicated techniques like limits and series and so on should be used to resolve the ambiguity.